### PR TITLE
Parsing branch

### PR DESCRIPTION
**Pull request by FjjDessoyCaraballo**
#### Parsing

On our parsing side we are extracting many variables that will be listed below:

1. `int map_start`: row in which map starts;
2. `int map_end`: row in which map ends;
3. `char **map`: array containing the map only;
4. `size_t map_width`: self-explanatory;
5. `size_t map_length`: self-explanatory;
6. `char **file`: array with all the contents from original file;
7. `char *floor_info`: array containing the whole string from the file detailing the floor RGB (not useful for gameplay);
8. `char *ceiling_info`: array containing the whole string from the file detailing the ceiling RGB (not useful for gameplay);
9. `size_t c_red/c_green/c_blue`: RGB values for the ceiling;
10. `size_t f_red/f_green/f_blue`: RGB values for the floor;
11. `char *n_sprite`: path to the wall sprite (north);
12. `char *s_sprite`: path to the wall sprite (south);
13. `char *e_sprite`: path to the wall sprite (east);
14. `char *w_sprite`: path to the wall sprite (west);
15. `bool n_player/s_player/w_player/e_player`: booleans that say if which position the player is facing;
16. `double x_ppos`: x player position (column);
17. `double y_ppos`: y player position (row);
18. `int repeat_test`: variable that is used solely for error_handling. Not for graphic side.

These are all extracted from the original file `.cub` and are available in the `t_data *data` struct.

Example on how to use:

```c
int8_t foo(t_data *struct)
{
	if (data->x_ppos == data->map_end)
		return (FAILURE);
}
```

### CODE REVIEW

File: inc/cubd.h [hunk 1]
```diff
@@ -38,41 +38,67 @@
...
+# define USAGE7 "cardinal orientation, proper walled map, and \n"
 # define USAGE8 "RGB notated instructions [numerical value]);\n\n"
 # define USAGE9 "This game has no exit for now, so use ESC to "
 # define USAGE10 "finish the game (:\n\n"
-# define USAGE11 "By: fdessoy- | araveala\n"
+# define USAGE11 "By: fdessoy- && araveala\n"
 
 /*************************************************/
 /* structs ***************************************/
 /*************************************************/
```
Comment (sparklebutt) [inline inc/cubd.h:73]: different variable names to mine, am now using yours, ppos, map_width and map _length (height might be more of a clear name)

**Review by sparklebutt — COMMENTED** [review]
We did a manual walk through so not much left to comment on review. 
questions.
1. The use of int8 /uint8 , is this just a matter of better practice, bare in mind minus numbers will be required for certain position calculations as far as i can tell at this stage. eg for direction of player view.
2.  We need a minimum maxium check for map size, we can test and base it on performance, eg 50 x 50, 100 x 100 ect. 
3.  have merged with mine, no bigger issues noticed.

File: inc/cubd.h (hunk 1 above)
Comment (FjjDessoyCaraballo) [reply inc/cubd.h:79]: I guess we talked about variable names last meeting and I think you managed to change them using the VS code global search. Closing conversation for final merge.

**Review by FjjDessoyCaraballo — COMMENTED** [review]
All good on my side

**Pull request merged**